src/lib/rest/rest.cpp

```diff
@@ -714,9 +773,12 @@ static int connectionTreat
     }
     LM_T(LmtUriParams, ("notifyFormat: '%s'", ciP->uriParam[URI_PARAM_NOTIFY_FORMAT].c_str()));
 
-    /* In the case an actual Fiware-ServicePath is found, the default will be overwritten */
-    ciP->httpHeaders.servicePath = DEFAULT_SERVICE_PATH;
     MHD_get_connection_values(connection, MHD_HEADER_KIND, httpHeaderGet, &ciP->httpHeaders);
+    if (!ciP->httpHeaders.servicePathReceived)
```

Comment:
Running this after the MHD_get_connection_values() instead of before seems to work and save a function call (and the lot of ifs inside that fucntion). I understand that when MHD_get_connection_values() returns, all the HTTP headers has been processed.

If you agree with this, mark it with NTC.